tc/sgx/trusted_worker_manager/enclave/signup_enclave.cpp

```diff
@@ -296,24 +455,29 @@ void CreateSignupReportData(const char* pOriginatorPublicKeyHash,
     // data structure because it is (64 bytes) larger than the SHA256
     // hash (32 bytes) but we zero it out first to ensure that it is
     // padded with known data.
-    Zero(pReportData, sizeof(*pReportData));
-    sgx_status_t ret = sgx_sha256_msg(reinterpret_cast<const uint8_t*>(hashString.c_str()),
-        static_cast<uint32_t>(hashString.size()),
-        reinterpret_cast<sgx_sha256_hash_t*>(pReportData));
+    Zero(report_data, sizeof(*report_data));
+    sgx_status_t ret = sgx_sha256_msg(reinterpret_cast<const uint8_t*>(hash_string.c_str()),
+        static_cast<uint32_t>(hash_string.size()),
```

Comment:
Shouldn't this be static_cast<uint64_t> since you are using 64 bits in report data

Reply:
As per sgx_sha256_msg function signature, size of input source should be of type 32 bits.
sgx_status_t sgx_sha256_msg(const uint8_t *p_src,
                                                uint32_t src_len,
                                                sgx_sha256_hash_t *p_hash);
 
the size of the resulting report will be 64 bits because the type _sgx_report_data_t internally has uint8_t vector of  64 bits.

Reply:
To verify could you print the contents of report data [0-63] and check if the values are as expected. Input source is Uint8 * and unit32 is length.

Reply:
Ok